components/zcash_address/src/kind/unified/address.rs

```diff
@@ -380,13 +410,16 @@ mod tests {
     fn duplicate_typecode() {
         // Construct and serialize an invalid UA. This must be done using private
         // methods, as the public API does not permit construction of such invalid values.
-        let ua = Address(vec![
-            Item::Data(Receiver::Sapling([1; 43])),
-            Item::Data(Receiver::Sapling([2; 43])),
-        ]);
-        let encoded = ua.to_jumbled_bytes(Address::MAINNET);
+        let ua = Address {
+            revision: Revision::R0,
+            receivers: vec![
+                Item::Data(Receiver::Sapling([1; 43])),
+                Item::Data(Receiver::Sapling([2; 43])),
+            ],
+        };
+        let encoded = ua.to_jumbled_bytes(Address::MAINNET_R0);
         assert_eq!(
-            Address::parse_internal(Address::MAINNET, &encoded[..]),
+            Address::parse_internal(Address::MAINNET_R0, &encoded[..]),
```

Comment:
(Just picking this test as an example)

I'm a little unhappy about the lack of Revision 1 tests. I think we'll be fine for now (particularly if my earlier suggestion to generate arbitrary revisions for proptests is accepted), but please open an issue for expanding testing to cover revision 1.

Reply:
Opened https://github.com/zcash/librustzcash/issues/1647

Reply:
I actually think it's essential to have tests covering the differences between R0 and R1.